### PR TITLE
Sample bot

### PR DESCRIPTION
**Pull request by vitsalis**

### CODE REVIEW

File: bots/php/round.php [hunk 1]
```diff
@@ -1,21 +1,77 @@
...
+                2 => [ 1, 0 ],
+                3 => [ 0, -1 ],
+                4 => [ -1, 0 ]
+            ];
+            $delta = [
+                1 => 'NORTH',
+                2 => 'EAST',
+                3 => 'SOUTH',
+                4 => 'WEST'
+            ];
```
Comment (dionyziz) [inline bots/php/round.php:22]: These are not deltas, it should be called something else.

File: tests/bots/php/bot.php [hunk 2]
```diff
@@ -0,0 +1,211 @@
+<?php
+    class BotTest extends UnitTestWithFixtures {
+        public function testSendGameRequest() {
+            ob_start();
+            include 'bots/php/game.php';
+            $response = json_decode( ob_get_clean() );
+
+            $this->assertTrue( $response !== false, 'The game request response must have valid json' );
```
Comment (dionyziz) [inline tests/bots/php/bot.php]: This should be `NULL` not `false`. Please make your tests red before making them green.

File: tests/bots/php/bot.php [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+            $this->assertTrue( empty( $response ), 'The game request response must be an empty array' );
+        }
+        public function testSendBotRequest() {
+            ob_start();
+            include 'bots/php/bot.php';
+            $response = json_decode( ob_get_clean() );
+
+            $this->assertTrue( $response !== false, 'The bot request response must have valid json' );
+
+            $this->assertTrue( is_object( $response ), 'The game request response must be an array' );
```
Comment (dionyziz) [inline tests/bots/php/bot.php]: This is not a game request; and this is not an array. It is an object in PHP and a dictionary in JSON. Objects and arrays are different things in PHP; dictionaries and arrays are different things in JSON.

File: tests/bots/php/bot.php [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+            $this->assertEquals( 3, count( get_object_vars( $response ) ), 'The bot request response must be an array of 3 entries' );
```
Comment (dionyziz) [inline tests/bots/php/bot.php]: similarly

File: tests/bots/php/bot.php [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+            $this->assertTrue( is_array( $response ), 'The game request response must be an array' );
+            $this->assertTrue( empty( $response ), 'The game request response must be an empty array' );
+        }
+        public function testSendBotRequest() {
+            ob_start();
+            include 'bots/php/bot.php';
+            $response = json_decode( ob_get_clean() );
+
+            $this->assertTrue( $response !== false, 'The bot request response must have valid json' );
```
Comment (dionyziz) [inline tests/bots/php/bot.php]: similarly

File: tests/bots/php/bot.php [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+            $response = $this->roundRequestAndGetResponse( [
+                'round' => 1,
+                'map' => json_encode( [] ),
+                'gameid' => 1,
+                'myid' => 1,
+                'W' => 1,
+                'H' => 1
+            ] );
+
+            $this->assertTrue( $response !== false, 'The round request response must have valid json' );
```
Comment (dionyziz) [inline tests/bots/php/bot.php]: similarly

File: tests/bots/php/bot.php [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+                    ]
+                ] ),
+                'gameid' => 1,
+                'myid' => 1,
+                'W' => 100,
+                'H' => 100
+            ] );
+
+            $this->assertEquals( 1, count( $response->intent ), 'When a user has a creature intent must have 1 field' );
+            $this->assertTrue( is_object( $response->intent[ 0 ] ), 'Intent must be an array of objects' );
```
Comment (dionyziz) [inline tests/bots/php/bot.php]: Similarly

File: tests/bots/php/bot.php [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+                    $validDirections = [ 'NORTH', 'WEST', 'SOUTH', 'EAST' ];
+                    $directionValid = array_search( $response->intent[ 0 ]->direction, $validDirections );
+                    $this->assertTrue( is_int( $directionValid ), 'Response must have a valid direction' );
+                    break;
+                case 'NONE':
+                    $this->assertEquals( 'NONE', $response->intent[ 0 ]->direction, 'When action is NONE direction should be NONE too' );
+                    break;
+            }
+        }
+        public function testMoveOutOfBounds() {
```
Comment (dionyziz) [inline tests/bots/php/bot.php]: I suggest this test be removed. The bot can assume the grader will always behave correctly.

File: tests/bots/php/bot.php (hunk 7 above)
Comment (vitsalis) [inline tests/bots/php/bot.php]: but intent is an array of objects..

**Pull request merged**